### DISCUSSION TITLE
How to add component to gameobject using a plugin?

### DISCUSSION THREAD

**Discussion by art0007i** [nesrak1/UABEA#371] [Q&A] [answered]
I wanted to write a plugin which will add a component to a GameObject. I managed to add a MonoBehavior to the asset infos but I'm a bit stumped on how to actually add it to the GameObject.

here's my original attempt but this doesn't even get past the if statement because the game object doesn't have a BaseValueField.
```cs
foreach (var item in selection.Where((item) => item.ClassId == (int)AssetClassID.GameObject))
{
    Console.WriteLine(item.HasValueField);
    if (item.BaseValueField == null) continue;

    var comp = item.BaseValueField["m_Component"];
    // Add another component here???
}
```

**Comment by nesrak1** [comment, marked as answer]
The BaseValueField name is a little misleading and that field will be removed entirely in the new UABEA update. Its main purpose was to hold on to user made edits (like if you import dump). To read the base field for the first time, you should use actually use `workspace.GetBaseField(container)`.
